tests/test_builder.py

```diff
@@ -290,6 +290,26 @@ def test_apply_sequence():
     start.add_position((0, 0)).apply(sequence).braket.local_emulator().run(1)
 
 
+def test_assign_errors():
+    with pytest.raises(TypeError):
+        (
+            start.add_position((0, 0))
+            .rydberg.rabi.amplitude.uniform.piecewise_linear(
+                [0.2, "rabi_dur_scanned", 0.2], [0.0, 10, 10, 0.0]
+            )
+            .batch_assign(rabi_dur_scanned=1.0)
```

Comment:
I think it would be good to pass a tuple/list here to accurately test the functionality reflected in this PR